### PR TITLE
Fix `clippy::pedantic`

### PR DESCRIPTION
**Pull request by devtomio**
## What does this PR address?

Fixes [`clippy::pedantic`](https://rust-lang.github.io/rust-clippy/master/)

## Before submitting:

- [x] Make sure to follow [GitHub's guide](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request) on creating PR.
- [x] Did you read through [contribution guidelines](https://github.com/railwayapp/nixpacks/blob/main/CONTRIBUTING.md)?
- [ ] Did your changes require updates to the documentation? Have you updated those accordingly?
- [x] Did you write tests to cover your changes? Did it pass locally when running `cargo test`?
- [x] Have you run `cargo fmt`, `cargo check` and `cargo clippy` locally to ensure that CI passes?

### CODE REVIEW

File: src/nixpacks/app.rs [hunk 1]
```diff
@@ -33,6 +33,7 @@ impl App {
     }
 
     /// Check if a file exists
+    #[must_use]
```
Comment (coffee-cup) [inline src/nixpacks/app.rs:36]: What is the reasoning for having `must_use` here?

**Review by coffee-cup — COMMENTED** [review]
I'm happy with most of these changes besides all the `must_use`'s. What is the reasoning for them? Would love if we didn't have to add them everywhere

**Comment by devtomio** [comment]
> I'm happy with most of these changes besides all the `must_use`'s. What is the reasoning for them? Would love if we didn't have to add them everywhere

Seems it has many false positives, I'll remove them.

**Comment by coffee-cup** [comment]
> must_use just means you must use the return value. Is that a bad thing?

It is not a bad thing I'd prefer a global rule for this instead of specifically adding `must_use` to a select set of functions. New functions are also likely to not get this attribute as outside contributors will be unaware that they need to add it.

**Review by coffee-cup — APPROVED** [review]

**Comment by coffee-cup** [comment]
Can you please fix the conflict in `node.rs`. Besdies that it looks good to merge.

**Pull request merged**